File: api/src/Persistence/Entities/Todo.cs
namespace api.Persistence.Entities;

public class Todo : BaseEntity
{
    public int UserId { get; set; }

    public int? WorkplaceId { get; set; }
    public Workplace? Workplace { get; set; }

    public string? Title { get; set; }
    public string? Description { get; set; }

    public bool IsCompleted { get; set; }
    public DateTime? CompletedAt { get; set; }
}

File: client/src/app/contracts/todo.ts
export interface TodoDetailModel {
  id: number;
  title: string | null;
  description: string | null;
  workplaceId: number | null;
  workplaceName: string | null;
  isCompleted: boolean;
  completedAt: string | null;
  createdAt: string;
  lastEditedAt: string | null;
}

export interface TodoCreateModel {
  title: string;
  description: string | null;
  workplaceId: number | null;
}

export interface TodoUpdateModel {
  id: number;
  title: string;
  description: string | null;
  workplaceId: number | null;
}
